cmd/fleet/main.go

```diff
@@ -184,3 +160,142 @@ func NewCommand(version string) *cobra.Command {
 	cmd.Flags().StringP("config", "c", "fleet-server.yml", "Configuration for Fleet Server")
 	return cmd
 }
+
+type fleetServer struct {
+	cfg *config.Config
+
+	es *es.Client
+	sv saved.CRUD
+
+	lock           sync.Mutex
+	profilerCancel context.CancelFunc
+	serverCancel   context.CancelFunc
+	errCh          chan error
+}
+
+// NewFleetServer creates the actual fleet server service.
+func NewFleetServer(ctx context.Context, cfg *config.Config) (*fleetServer, error) {
+	es, err := es.New(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	sv := saved.NewMgr(es.Bulk(), savedObjectKey())
+	return &fleetServer{
+		cfg: cfg,
+		es:  es,
+		sv:  sv,
+	}, nil
+}
+
+// Run runs the fleet server.
+func (f *fleetServer) Run(ctx context.Context) error {
+	// Initial indices bootstrapping, needed for agents actions development
+	// TODO: remove this after the indices bootstrapping logic implemented in ES plugin
+	err := esboot.EnsureESIndices(ctx, f.es)
+	if err != nil {
+		return err
+	}
+
```

Comment:
maybe pefer
```
defer profilerCancel() 
```

Reply:
no don't want to defer that as, it can be replaced with a different cancellable on reload